multiversx_sdk/native_auth/native_auth_server.py

```diff
@@ -0,0 +1,284 @@
+
+        user_verifier = UserPublicKey(address.get_public_key())
+        return user_verifier.verify(serialized_message, signature)
+
+    def _get_current_block_timestamp(self) -> int:
+        if self.config.cache:
+            timestamp = self.config.cache.get("block:timestamp:latest")
+            if timestamp:
+                return int(timestamp)
+
+        response = requests.get(f"{self.config.api_url}/blocks?size=1&fields=timestamp")
+        timestamp = int(response.json()[0]["timestamp"])
+
+        if self.config.cache:
+            self.config.cache.set("block:timestamp:latest", timestamp, 6)
+
+        return timestamp
+
+    def _get_block_timestamp(self, hash: str) -> Union[int, None]:
+        if self.config.cache:
+            timestamp = self.config.cache.get(f"block:timestamp:{hash}")
+            if timestamp:
+                return int(timestamp)
+
+        try:
+            timestamp = requests.get(f"{self.config.api_url}/blocks/{hash}?extract=timestamp")
+
+            if timestamp.status_code == 404:
+                return None
+
+            timestamp = timestamp.text
+            if self.config.cache:
+                self.config.cache.set(f"block:timestamp:{hash}", int(timestamp), self.config.max_expiry_seconds)
+
+            return int(timestamp)
+        except Exception as ex:
+            raise ex
+
+    def _is_origin_accepted(self, origin: str) -> bool:
+        if self._is_wildcard_origin_accepted(origin):
+            return True
+
+        is_accepted = origin in self.config.accepted_origins or f"https://{origin}" in self.config.accepted_origins
+        if is_accepted:
+            return True
+
+        if self.config.is_origin_accepted:
+            return self.config.is_origin_accepted(origin)
+
+        return False
+
+    def _is_wildcard_origin_accepted(self, origin: str) -> bool:
+        if origin in self.accepted_wildcard_origins:
+            return True
+
+        if not len(self.wildcard_origins):
+            return False
+
+        wildcard_origin = [
+            wildcard
+            for wildcard in self.wildcard_origins
+            if origin.startswith(wildcard.protocol) and origin.endswith(wildcard.domain)
+        ]
+
+        if not wildcard_origin:
+            return False
+
+        # append new origin to the end of the list
+        self.accepted_wildcard_origins.append(origin)
+
+        if len(self.accepted_wildcard_origins) > 1000:
+            # remove the first element of the list which is the oldest
+            self.accepted_wildcard_origins.pop(0)
+
+        return True
+
+    def _decode_value(self, value: str) -> str:
+        return base64.urlsafe_b64decode(self._ensure_correct_base64_padding(value)).decode("utf-8")
+
+    def _ensure_correct_base64_padding(self, value: str) -> str:
+        missing_padding = len(value) % 4
+        if missing_padding:
+            value += "=" * (4 - missing_padding)
+        return value
```

Comment:
Does decoding with `validate=False` help, perhaps (instead of coding the additional padding logic)?

Reply:
as discussed, this is ok, will stay as it is.